website/content/en/docs/reference/threat-model.md

```diff
@@ -29,92 +28,161 @@ A Cluster Developer has the ability to create pods via `Deployments`, `ReplicaSe
+
+Mitigation: Access to the `NodePool` is treated as a privileged permission. If an Application Owners had access to the `NodePool`, they could modify or delete everything about the existing nodes in the cluster. By default, only the Cluster Operator has access to this resource. If the Cluster Operator grants another user permission, this is the user’s responsibility. Karpenter also emits a metric for the current allowed disruption that users can monitor in case their `NodePools` are modified. Karpenter will also emit an event for the `NodePool` if disruptions are allowed. 
+
+#### Launching EC2 instances with IAM roles not intended for Karpenter nodes
 
-### Threat: Karpenter can be used to create or terminate EC2 instances outside the cluster
+**Threat:** Many IAM roles in an AWS account may trust the EC2 service principal. IAM administrators must grant the `iam:PassRole` permission to `IAM principals` to allow those principals in the account to launch instances with specific roles. A Cluster Operator attempts to create an `EC2NodeClass` with an IAM role not intended for Karpenter
 
-**Background**: EC2 instances can exist in an AWS account outside of the Kubernetes cluster.
+**Mitigation:** Cluster Operators must enumerate the roles in the resource section of the IAM policy granted to the Karpenter role for the `iam:PassRole` action. Karpenter will fail to generate an instance profile if role that is specified in the `spec.role` section of the `EC2NodeClass` is not enumerated in the `iam:PassRole` permission.
 
-**Threat**: An actor who obtains control of the Karpenter pod’s IAM role may be able to create or terminate EC2 instances not part of the Kubernetes cluster managed by Karpenter.
+#### Execution of Arbitrary UserData
 
-**Mitigation**: Karpenter creates instances with tags, several of which are enforced in the IAM policy granted to the Karpenter IAM role that restrict the instances Karpenter can terminate. One tag requires that the instance was provisioned by a Karpenter controller (`karpenter.sh/nodepool`), another tag can include a cluster name to mitigate any termination between two clusters with Karpenter in the same account (`kubernetes.io/cluster/${CLUSTER_NAME}`. Cluster Operators also can restrict the region to prevent two clusters in the same account with the same name in different regions.
+**Threat:** Karpenter allows users to specify custom UserData as part of the `EC2NodeClass` resource. This UserData may take two shapes: a MIME multi-part archive or a NodeConfig yaml object. Karpenter does not perform any validation on this input, other than parsing the MIME multi-part archive if a header is detected. This custom UserData is then merged with UserData generated by Karpenter and is included in the Launch Template used to launch Karpenter managed nodes. Users with EDIT permissions on `EC2NodeClass` are able arbitrarily execute code on nodes launched by Karpenter.
```

Comment:
Can we reference out to our userData section of our EC2NodeClass doc where we state that we support these formats? Also, this isn't true for all forms of AMIs so it may not be necessary mentioning the format at all here